{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# RNA Seq Analysis - Template"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import matplotlib.pyplot as plt \n",
    "import matplotlib.style as style\n",
    "style.use('ggplot')\n",
    "#style.use('Solarize_Light2')\n",
    "from matplotlib import rcParams\n",
    "rcParams['pdf.fonttype'] = 42 ## Output Type 3 (Type3) or Type 42 (TrueType)\n",
    "rcParams['font.sans-serif'] = 'Arial'\n",
    "from pylab import *\n",
    "\n",
    "# ignore FutureWarning that may pop up when plotting\n",
    "import warnings\n",
    "warnings.filterwarnings(\"ignore\")\n",
    "import urllib3\n",
    "urllib3.disable_warnings()\n",
    "\n",
    "from dfply import *\n",
    "\n",
    "# For PCA\n",
    "from sklearn.decomposition import PCA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "# load preprocessed harmonized data from local file\n",
    "genes_with_meta = pd.read_csv(\"small_Harmonized_rna_seq_data.csv\")\n",
    "genes_with_meta.head(2)\n",
    "\n",
    "#Clean up data format\n",
    "genes_with_meta = genes_with_meta.replace(to_replace = \"male\", value = \"Male\")\n",
    "genes_with_meta = genes_with_meta.replace(to_replace = \"female\", value = \"Female\")\n",
    "genes_with_meta = genes_with_meta.rename(columns={'used': 'id'})\n",
    "genes_with_meta = genes_with_meta.reset_index()\n",
    "genes_with_meta = genes_with_meta.drop(labels=[\"index\", \"path\"], axis =1)\n",
    "\n",
    "#Quality control\n",
    "#removing any rows or columns with uncompliant values\n",
    "genes_with_meta = genes_with_meta[genes_with_meta.totalCounts != \"Male\"]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Data file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>ROW_ID</th>\n",
       "      <th>ROW_VERSION</th>\n",
       "      <th>totalCounts</th>\n",
       "      <th>Symbol</th>\n",
       "      <th>zScore</th>\n",
       "      <th>specimenID</th>\n",
       "      <th>parent</th>\n",
       "      <th>individualID</th>\n",
       "      <th>assay</th>\n",
       "      <th>...</th>\n",
       "      <th>resourceType</th>\n",
       "      <th>nf1Genotype</th>\n",
       "      <th>nf2Genotype</th>\n",
       "      <th>studyName</th>\n",
       "      <th>id</th>\n",
       "      <th>age</th>\n",
       "      <th>isCellLine</th>\n",
       "      <th>experimentalCondition</th>\n",
       "      <th>transplantationType</th>\n",
       "      <th>modelOf</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>9324</td>\n",
       "      <td>136.993</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>-0.226861</td>\n",
       "      <td>patient10tumor1</td>\n",
       "      <td>syn18407530</td>\n",
       "      <td>patient10</td>\n",
       "      <td>rnaSeq</td>\n",
       "      <td>...</td>\n",
       "      <td>experimentalData</td>\n",
       "      <td>-/-</td>\n",
       "      <td>unknown</td>\n",
       "      <td>Cutaneous Neurofibroma Data Resource</td>\n",
       "      <td>syn20430744</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Cutaneous Neurofibroma</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>9324</td>\n",
       "      <td>168.454</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>-0.248889</td>\n",
       "      <td>patient10tumor2</td>\n",
       "      <td>syn18407530</td>\n",
       "      <td>patient10</td>\n",
       "      <td>rnaSeq</td>\n",
       "      <td>...</td>\n",
       "      <td>experimentalData</td>\n",
       "      <td>-/-</td>\n",
       "      <td>unknown</td>\n",
       "      <td>Cutaneous Neurofibroma Data Resource</td>\n",
       "      <td>syn20430745</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Cutaneous Neurofibroma</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "      <td>9324</td>\n",
       "      <td>115.456</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>-0.115625</td>\n",
       "      <td>patient10tumor3</td>\n",
       "      <td>syn18407530</td>\n",
       "      <td>patient10</td>\n",
       "      <td>rnaSeq</td>\n",
       "      <td>...</td>\n",
       "      <td>experimentalData</td>\n",
       "      <td>-/-</td>\n",
       "      <td>unknown</td>\n",
       "      <td>Cutaneous Neurofibroma Data Resource</td>\n",
       "      <td>syn20430746</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Cutaneous Neurofibroma</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>4</td>\n",
       "      <td>9324</td>\n",
       "      <td>500.304</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>-0.184728</td>\n",
       "      <td>patient11tumor1</td>\n",
       "      <td>syn18407530</td>\n",
       "      <td>patient11</td>\n",
       "      <td>rnaSeq</td>\n",
       "      <td>...</td>\n",
       "      <td>experimentalData</td>\n",
       "      <td>-/-</td>\n",
       "      <td>unknown</td>\n",
       "      <td>Cutaneous Neurofibroma Data Resource</td>\n",
       "      <td>syn20430747</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Cutaneous Neurofibroma</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>5</td>\n",
       "      <td>9324</td>\n",
       "      <td>380.698</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>-0.229165</td>\n",
       "      <td>patient11tumor14</td>\n",
       "      <td>syn18407530</td>\n",
       "      <td>patient11</td>\n",
       "      <td>rnaSeq</td>\n",
       "      <td>...</td>\n",
       "      <td>experimentalData</td>\n",
       "      <td>-/-</td>\n",
       "      <td>unknown</td>\n",
       "      <td>Cutaneous Neurofibroma Data Resource</td>\n",
       "      <td>syn20430748</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Cutaneous Neurofibroma</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 28 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   Unnamed: 0  ROW_ID  ROW_VERSION  totalCounts Symbol    zScore  \\\n",
       "0           0       1         9324      136.993   A1BG -0.226861   \n",
       "1           1       2         9324      168.454   A1BG -0.248889   \n",
       "2           2       3         9324      115.456   A1BG -0.115625   \n",
       "3           3       4         9324      500.304   A1BG -0.184728   \n",
       "4           4       5         9324      380.698   A1BG -0.229165   \n",
       "\n",
       "         specimenID       parent individualID   assay  ...      resourceType  \\\n",
       "0   patient10tumor1  syn18407530    patient10  rnaSeq  ...  experimentalData   \n",
       "1   patient10tumor2  syn18407530    patient10  rnaSeq  ...  experimentalData   \n",
       "2   patient10tumor3  syn18407530    patient10  rnaSeq  ...  experimentalData   \n",
       "3   patient11tumor1  syn18407530    patient11  rnaSeq  ...  experimentalData   \n",
       "4  patient11tumor14  syn18407530    patient11  rnaSeq  ...  experimentalData   \n",
       "\n",
       "  nf1Genotype nf2Genotype                             studyName           id  \\\n",
       "0         -/-     unknown  Cutaneous Neurofibroma Data Resource  syn20430744   \n",
       "1         -/-     unknown  Cutaneous Neurofibroma Data Resource  syn20430745   \n",
       "2         -/-     unknown  Cutaneous Neurofibroma Data Resource  syn20430746   \n",
       "3         -/-     unknown  Cutaneous Neurofibroma Data Resource  syn20430747   \n",
       "4         -/-     unknown  Cutaneous Neurofibroma Data Resource  syn20430748   \n",
       "\n",
       "  age isCellLine experimentalCondition transplantationType  \\\n",
       "0 NaN        NaN                   NaN                 NaN   \n",
       "1 NaN        NaN                   NaN                 NaN   \n",
       "2 NaN        NaN                   NaN                 NaN   \n",
       "3 NaN        NaN                   NaN                 NaN   \n",
       "4 NaN        NaN                   NaN                 NaN   \n",
       "\n",
       "                  modelOf  \n",
       "0  Cutaneous Neurofibroma  \n",
       "1  Cutaneous Neurofibroma  \n",
       "2  Cutaneous Neurofibroma  \n",
       "3  Cutaneous Neurofibroma  \n",
       "4  Cutaneous Neurofibroma  \n",
       "\n",
       "[5 rows x 28 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "genes_with_meta.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The following table contains detailed descriptions of the metadata variables included in the data file:\n",
    "\n",
    "Var | Description\n",
    "--- | --- \n",
    " _id_  |  the synapseIDs of the individual files with raw data\n",
    " _specimenID_  |  individual samples\n",
    " _species_  |  the source of the specimen\n",
    " _age_  |  the age of the patient \n",
    " _Sex_  |  the sex of the patient\n",
    " _tumorType_  |  the the type of tumor, can be one of 7 different diagnoses\n",
    " _isCellLine_  |  indicates whether the origin tissue was a cell line or a patient\n",
    " _study_  |  the specific initiative/consortia that the study was a part of\n",
    " _Symbol_  |  the common names of genes \n",
    " _cellCulture_  |  whether the data was obtained from a cell culture experiment or an _in-vivo_ experiment. Many inherent differences exist between _cell culture_ experiments and _in-vivo_ experiments. These differences may need to be taken into account during analysis and interpretation of the data\n",
    " _totalCounts_  |  the main measure of expression levels of the gene. _(the values documented here are not corrected for batch/study)_ \n",
    " _zScore_  |  normalized counts (counts for a gene can vary significantly due to technical differences in different studies, so we normalized the counts into a zScore for a comparable measurement between samples for the same gene)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Metadata"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
